rocketpy/simulation/monte_carlo.py

```diff
@@ -181,186 +220,406 @@ def simulate(
         self.number_of_simulations = number_of_simulations
-        self.__iteration_count = self.num_of_loaded_sims if append else 0
-        self.__start_time = time()
-        self.__start_cpu_time = process_time()
+        self._initial_sim_idx = self.num_of_loaded_sims if append else 0
 
         # Begin display
-        print("Starting Monte Carlo analysis", end="\r")
+        MonteCarlo._reprint("Starting Monte Carlo analysis")
 
-        try:
-            while self.__iteration_count < self.number_of_simulations:
-                self.__run_single_simulation(input_file, output_file)
-        except KeyboardInterrupt:
-            print("Keyboard Interrupt, files saved.")
-            error_file.write(json.dumps(self._inputs_dict, cls=RocketPyEncoder) + "\n")
-            self.__close_files(input_file, output_file, error_file)
-        except Exception as error:
-            print(f"Error on iteration {self.__iteration_count}: {error}")
-            error_file.write(json.dumps(self._inputs_dict, cls=RocketPyEncoder) + "\n")
-            self.__close_files(input_file, output_file, error_file)
-            raise error
-        finally:
-            self.total_cpu_time = process_time() - self.__start_cpu_time
-            self.total_wall_time = time() - self.__start_time
+        # Setup files
+        self.__setup_files(append)
 
-        self.__terminate_simulation(input_file, output_file, error_file)
+        # Run simulations
+        if parallel:
+            self.__run_in_parallel(n_workers)
+        else:
+            self.__run_in_serial()
 
-    # Auxiliary methods
+        self.__terminate_simulation()
 
-    def __run_single_simulation(self, input_file, output_file):
+    def __setup_files(self, append):
         """
-        Runs a single simulation and saves the inputs and outputs to the
-        respective files.
+        Sets up the files for the simulation, creating then if necessary.
 
         Parameters
         ----------
-        input_file : str
-            The file object to write the inputs.
-        output_file : str
-            The file object to write the outputs.
+        append : bool
+            If ``True``, the results will be appended to the existing files. If
+            ``False``, the files will be overwritten.
 
         Returns
         -------
         None
         """
-        self.__iteration_count += 1
-
-        monte_carlo_flight = Flight(
-            rocket=self.rocket.create_object(),
-            environment=self.environment.create_object(),
-            rail_length=self.flight._randomize_rail_length(),
-            inclination=self.flight._randomize_inclination(),
-            heading=self.flight._randomize_heading(),
-            initial_solution=self.flight.initial_solution,
-            terminate_on_apogee=self.flight.terminate_on_apogee,
-        )
+        # Create data files for inputs, outputs and error logging
+        open_mode = "r+" if append else "w+"
```

Comment:
The stream is positioned in the beginning of the file solely for the purpose of the `readlines()` method (if it were at the end, it always return 0). I guess this check of `id_i` and `id_o` might be optional? What is your suggestion here?

The issue is that the `MonteCarlo` class only knows whether it should overwrite or append to the files when the `simulate` method is called, since it has the `append` parameter.